### PR TITLE
[#4927] improvement(catalog): extract a common HMS module

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

extract a common HMS module

### Why are the changes needed?

Fix: #4927 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

CI passed

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -97,6 +97,11 @@ mail = "1.4.1"
 rome = "1.0"
 jettison = "1.1"
 thrift = "0.12.0"
+derby = "10.14.2.0"
+datanucleus-core = "4.1.17"
+datanucleus-api-jdo = "4.2.4"
+datanucleus-rdbms = "4.1.19"
+datanucleus-jdo = "3.2.0-m3"
```
Comment (jerryshao) [inline gradle/libs.versions.toml:104]: Are they just used for test? If not, you should also update the license.bin.

File: catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveTable.java [hunk 2]
```diff
@@ -76,55 +72,15 @@ private HiveTable() {}
...
-    StorageDescriptor sd = table.getSd();
-    Distribution distribution = Distributions.NONE;
-    if (sd.getBucketCols() != null && !sd.getBucketCols().isEmpty()) {
-      // Hive table use hash strategy as bucketing strategy
-      distribution =
-          Distributions.hash(
-              sd.getNumBuckets(),
-              sd.getBucketCols().stream().map(NamedReference::field).toArray(Expression[]::new));
-    }
+    Distribution distribution = ConvertUtil.getDistribution(table);
```
Comment (jerryshao) [inline catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveTable.java:77]: `CovertUtil` seems not so reasonable, maybe "HiveUtil" or "HiveTableUtil", "HiveTableConverter"?

File: catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestHiveTableOperations.java [hunk 3]
```diff
@@ -31,8 +31,8 @@
 import java.util.Arrays;
 import java.util.Map;
 import org.apache.gravitino.NameIdentifier;
-import org.apache.gravitino.catalog.hive.miniHMS.MiniHiveMetastoreService;
 import org.apache.gravitino.exceptions.NoSuchPartitionException;
+import org.apache.gravitino.hms.miniHMS.MiniHiveMetastoreService;
```
Comment (jerryshao) [inline catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestHiveTableOperations.java:35]: Can we change the package name to "o.a.g.hive.hms.xxx", it is weird that the package name has uppercase letters...

File: catalogs/hive-metastore-common/src/main/java/org/apache/gravitino/hms/dyn/DynMethods.java [hunk 4]
```diff
@@ -16,7 +16,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.gravitino.catalog.hive.dyn;
+package org.apache.gravitino.hms.dyn;
```
Comment (jerryshao) [inline catalogs/hive-metastore-common/src/main/java/org/apache/gravitino/hms/dyn/DynMethods.java:19]: I would change to "xxx.hive.dyn"

File: gradle/libs.versions.toml (hunk 1 above)
Comment (mchades) [reply gradle/libs.versions.toml:104]: I saw they are already in.
https://github.com/apache/gravitino/blob/4f961bb2d498fb6158b4f1db0b249e8a1b356149/LICENSE.bin#L323

https://github.com/apache/gravitino/blob/4f961bb2d498fb6158b4f1db0b249e8a1b356149/LICENSE.bin#L279

File: catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveTable.java (hunk 2 above)
Comment (mchades) [reply catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveTable.java]: renamed to `HiveTableConverter`

File: catalogs/hive-metastore-common/src/main/java/org/apache/gravitino/hms/dyn/DynMethods.java (hunk 4 above)
Comment (mchades) [reply catalogs/hive-metastore-common/src/main/java/org/apache/gravitino/hms/dyn/DynMethods.java]: fixed

File: catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestHiveTableOperations.java (hunk 3 above)
Comment (mchades) [reply catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestHiveTableOperations.java]: renamed

**Review by jerryshao — APPROVED** [review]
LGTM.

**Pull request merged**